src/gui/sidebar/previews/page/SidebarPreviewPages.cpp

```diff
@@ -11,13 +11,53 @@
 
 SidebarPreviewPages::SidebarPreviewPages(Control* control, GladeGui* gui, SidebarToolbar* toolbar)
  : SidebarPreviewBase(control, gui, toolbar)
+ , contextMenu(gui->get("sidebarPreviewContextMenu"))
 {
 	XOJ_INIT_TYPE(SidebarPreviewPages);
+
+	// Connect the context menu actions
+	const std::map<std::string, SidebarActions> ctxMenuActions = {
+	        {"sidebarPreviewDuplicate", SIDEBAR_ACTION_COPY},
+	        {"sidebarPreviewDelete", SIDEBAR_ACTION_DELETE},
+	        {"sidebarPreviewMoveUp", SIDEBAR_ACTION_MOVE_UP},
+	        {"sidebarPreviewMoveDown", SIDEBAR_ACTION_MOVE_DOWN},
+	        {"sidebarPreviewNewBefore", SIDEBAR_ACTION_NEW_BEFORE},
+	        {"sidebarPreviewNewAfter", SIDEBAR_ACTION_NEW_AFTER},
+	};
+
+	for (const auto& pair: ctxMenuActions)
+	{
+		GtkWidget* const entry = gui->get(pair.first);
+		g_assert(entry != nullptr);
+
+		// Unfortunately, we need a fairly complicated mechanism to keep track
+		// of which action we want to execute.
+		typedef SidebarPreviewPages::ContextMenuData Data;
+		auto userdata = std::unique_ptr<Data>(new Data{this->toolbar, pair.second});
+
+		const auto callback =
+		        G_CALLBACK(+[](GtkMenuItem* item, Data* data) { data->toolbar->runAction(data->actions); });
+		const gulong signalId = g_signal_connect(entry, "activate", callback, userdata.get());
+		g_object_ref(entry);
+		this->contextMenuSignals.push_back(std::make_tuple(entry, signalId, std::move(userdata)));
```

Comment:
Try to use `this->contextMenuSignals.emplace_back(entry, signalId, std::move(userdata));`